app/ota_client.py

```diff
@@ -257,91 +230,99 @@ def _header_str_to_dict(header_str):
     return header_dict
 
 
-def _save_update_ecuinfo(update_ecuinfo_yaml_file, update_ecu_info):
+def _save_update_ecuinfo(update_ecuinfo_yaml_file: Path, update_ecu_info: Path):
     """
     save update ecuinfo.yaml
     """
     output_file = update_ecuinfo_yaml_file
     logger.info(f"output_file: {output_file}")
-    with tempfile.NamedTemporaryFile("w", delete=False) as ftmp:
+    with tempfile.NamedTemporaryFile("w", delete=False, prefix=__name__) as ftmp:
         tmp_file_name = ftmp.name
-        with open(tmp_file_name, "w") as f:
-            f.write(yaml.dump(update_ecu_info))
-            f.flush()
+        ftmp.write(yaml.dump(update_ecu_info))
+
     shutil.move(tmp_file_name, output_file)
-    os.sync()
     return True
 
 
-class DirectoryInf:
+class _BaseInf:
+    _base_pattern = re.compile(
+        r"(?P<mode>\d+),(?P<uid>\d+),(?P<gid>\d+),(?P<left_over>.*)"
+    )
+
+    def __init__(self, info: str):
+        match_res: re.Match = self._base_pattern.match(info.strip("\n"))
+        assert match_res is not None
+        self.mode = int(match_res.group("mode"), 8)
+        self.uid = int(match_res.group("uid"))
+        self.gid = int(match_res.group("gid"))
+
+        self._left: str = match_res.group("left_over")
+
+
+class DirectoryInf(_BaseInf):
     """
     Directory file information class
     """
 
     def __init__(self, info):
-        line = info.replace("\n", "")
-        info_list, last = _get_separated_strings(line, ",", 3)
-        self.mode = info_list[0]
-        self.uid = info_list[1]
-        self.gpid = info_list[2]
-        self.path = _decapsulate(line[last:])
+        super().__init__(info)
+        self.path = Path(self._left[1:-1])
 
 
-class SymbolicLinkInf:
+class SymbolicLinkInf(_BaseInf):
     """
     Symbolik link information class
     """
 
+    _pattern = re.compile(r"'(?P<link>.+)','(?P<target>.+)'")
```

Comment:
It seems that the following path can not be parsed:
```
'/var/lib/ieee-data/iab.csv','/usr/share/ieee-data/'\'','\''iab.csv'
```
original file name contains `','` and escaped by `'\'','\''`

Reply:
I will take a look at it.

Reply:
Fix in 9b66362.
I introduce a method `regulate_escape` to replace all `'\''` to `\'` literally before doing regex.
pathlib.Path can handle `\'` properly itself.

Reply:
Do you mean that the original file name without any escaping is `/usr/share/ieee-data/','iab.csv`?

Reply:
> Do you mean that the original file name without any escaping is /usr/share/ieee-data/','iab.csv?

Yes!

Reply:
should be fix in a113294.
I also add this special case into the test suits!